src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -972,6 +973,400 @@ static void Main()
+                        Identical([string.Empty]); // Span<string>
+                        SpanDerived([string.Empty]); // Span<string>
+                    }
+                }
+                """;
+            var comp = CreateCompilation(
+                new[] { sourceA, sourceB1 },
+                targetFramework: TargetFramework.Net80,
+                options: TestOptions.ReleaseExe);
+            CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: IncludeExpectedOutput("""
+                T[]
+                string[]
+                string[]
+                Span<T>
+                Span<string>
+                Span<string>
+                """));
+
+            string sourceB2 = """
+                partial class Program
+                {
+                    static void Main()
+                    {
+                        SpanDerived(new[] { string.Empty }); // ambiguous
+                        ArrayDerived([string.Empty]); // ambiguous
+                    }
+                }
+                """;
+            comp = CreateCompilation(
+                new[] { sourceA, sourceB2 },
+                targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // 1.cs(5,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.SpanDerived(Span<string>)' and 'Program.SpanDerived(object[])'
+                //         SpanDerived(new[] { string.Empty }); // ambiguous
+                Diagnostic(ErrorCode.ERR_AmbigCall, "SpanDerived").WithArguments("Program.SpanDerived(System.Span<string>)", "Program.SpanDerived(object[])").WithLocation(5, 9),
+                // 1.cs(6,9): error CS0121: The call is ambiguous between the following methods or properties: 'Program.ArrayDerived(Span<object>)' and 'Program.ArrayDerived(string[])'
+                //         ArrayDerived([string.Empty]); // ambiguous
+                Diagnostic(ErrorCode.ERR_AmbigCall, "ArrayDerived").WithArguments("Program.ArrayDerived(System.Span<object>)", "Program.ArrayDerived(string[])").WithLocation(6, 9));
+        }
+
+        [Fact]
+        public void BetterConversionFromExpression_03()
+        {
+            string sourceA = """
+                using System;
+                using static System.Console;
+
+                partial class Program
+                {
+                    static void Unrelated(Span<int> value) { WriteLine("Span<int>"); }
+                    static void Unrelated(string[] value)     { WriteLine("string[]"); }
+                }
+                """;
+
+            string sourceB1 = """
+                partial class Program
+                {
+                    static void Main()
+                    {
+                        Unrelated(new[] { 1 }); // Span<int>
+                        Unrelated(new[] { string.Empty }); // string[]
+
+                        Unrelated([2]); // Span<string>
```

Comment:
`Span<int>`?

Reply:
Yes, thanks. I'll fix up in the next PR.